### ISSUE TITLE
[Tahap 9C] Fine-tuning dan Evaluasi Model IndoBERT

### ISSUE THREAD

**Issue by pramacoder** [pramacoder/Analisis-Sentimen-Cakap-Berbasis-Aspek#9]
## Deskripsi
Fine-tuning model pretrained IndoBERT untuk Task A (Aspect Classification) dan Task B (Sentiment Classification per Aspect).

## Model Pretrained
- **Model:** `indobenchmark/indobert-base-p2`
- **Tokenizer:** WordPiece (subword BPE)
- **Source:** HuggingFace Transformers

## Arsitektur Fine-tuning

```
Input Text (subword tokenizer BPE)
    ↓
IndoBERT Pretrained (indobert-base-p2)
    ↓
[CLS] Token Representation
    ↓
Dropout (0.1)
    ↓
Linear Classifier Head
    ↓
Output Label (aspek / sentimen)
```

**Keunggulan:** Memahami konteks bahasa Indonesia informal, singkatan, typo, dan frasa ambigu yang sering muncul di ulasan aplikasi.

## Hyperparameter

| Parameter | Nilai |
|---|---|
| Learning Rate | 2e-5 |
| Batch Size | 16 |
| Max Epochs | 5–10 |
| Dropout | 0.1 |
| Optimizer | AdamW |
| Warmup Steps | 10% dari total training steps |
| Max Token Length | 128 |
| Weight Decay | 0.01 |

## Kebutuhan Komputasi
Gunakan GPU untuk fine-tuning. Rekomendasi:
- Google Colab Pro (T4 / A100)
- Kaggle Notebook (T4 x2, gratis 30 jam/minggu)

### Setup Environment
- [ ] Install `transformers`, `datasets`, `torch`
- [ ] Load tokenizer `indobert-base-p2`
- [ ] Tokenisasi dataset dengan padding & truncation (max_length=128)

### Task A — Aspect Classification
- [ ] Setup DataLoader dengan IndoBERT tokenizer
- [ ] Fine-tuning 5–10 epoch
- [ ] Evaluasi: Accuracy, F1-Macro, Confusion Matrix
- [ ] Simpan checkpoint terbaik

### Task B — Sentiment Classification per Aspect
- [ ] Fine-tuning untuk 3-class sentiment
- [ ] Evaluasi per aspek
- [ ] Generate confusion matrix

### Finalisasi
- [ ] Simpan model fine-tuned di `models/indobert_finetuned/`
- [ ] Ekspor hasil ke `results/indobert_results.csv`

## Output
- [ ] `notebooks/07_indobert.ipynb`
- [ ] `models/indobert_finetuned/` (config + weights)
- [ ] `results/indobert_results.csv`
- [ ] Grafik training curves di `figures/`
